include/gadget/common.h

```diff
@@ -0,0 +1,35 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef __COMMON_H
+#define __COMMON_H
+
+#include <vmlinux.h>
+#include <gadget/types.h>
+#include <gadget/mntns.h>
+#include <bpf/bpf_helpers.h>
+
+// gadget_fill_current_process fills the given process struct with the current
+// process information.
+void static __always_inline gadget_fill_current_process(struct gadget_process *p) {
+	__u64 pid_tgid = bpf_get_current_pid_tgid();
+	__u64 uid_gid = bpf_get_current_uid_gid();
+
+	p->pid = pid_tgid >> 32;
+	p->tid = pid_tgid;
+	bpf_get_current_comm(p->comm, sizeof(p->comm));
+
+	p->user.uid = uid_gid;
+	p->user.gid = uid_gid >> 32;
+
+	struct task_struct *task = (struct task_struct *)bpf_get_current_task();
+	p->mntns_id = BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+
+	struct task_struct *parent = BPF_CORE_READ(task, real_parent);
+	if (parent == NULL)
+		return;
+
+	bpf_probe_read_kernel(&p->parent.comm, sizeof(p->parent.comm), parent->comm);
+	p->parent.pid = BPF_CORE_READ(parent, tgid);
+}
```

Comment:
To check:
- Can we get uid/gid and pid/tid directly from task_struct to avoid calling those two helpers. Does it provide any performance advantage?
- Should we provide a version of this without using the BPF_CORE_READ macro nor bpf_probe_read_lernel for tracing programs?

Reply:
I think it's better to stay with the helpers.